### PR TITLE
#4288 Generic signer modal

### PR DESCRIPTION
**Pull request by desfero**
TODO:
- [x] Write unit test for BaseAnimation component

Android: https://i.diawi.com/Yc3q8c
IOS: https://i.diawi.com/zJxXBh

### CODE REVIEW

File: packages/wallet/app/modules/signer-ui/reducer.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { AppReducer } from "@neufund/sagas";
 
 import { signerUIActions } from "./actions";
-import { TSignerRequestData } from "./types";
+import { TSignerSignPayload } from "./types";
 
 export enum ESignerUIState {
   IDLE = "idle",
```
Comment (arjunu) [inline packages/wallet/app/modules/signer-ui/reducer.ts:7]: do you think "hidden" is more appropriate than "idle"?

**Comment by arjunu** [comment]
- there's a typo in the file name `BottomSheetModa.stories.tsx` and it fails to load 
image

- can you add stories for the three types of modal?

File: packages/wallet/storybook/stories.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import "../app/components/shared/LineBreak.stories";
 import "../app/components/shared/Icon.stories";
 import "../app/components/shared/Link.stories";
 import "../app/components/shared/asset/Asset.stories";
+import "../app/components/shared/modals/BottomSheetModa.stories";
```
Comment (arjunu) [inline packages/wallet/storybook/stories.ts:7]: BottomSheetModal*

File: packages/wallet/app/components/shared/animations/BaseAnimation.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    if (prevProps.isActive !== this.props.isActive) {
+      if (this.props.isActive) {
+        this.showAnimation();
+      } else {
+        this.hideAnimation();
+      }
+    }
+  }
+
+  showAnimation() {
```
Comment (arjunu) [inline packages/wallet/app/components/shared/animations/BaseAnimation.tsx:65]: `startAnimation` and `finishAnimation` seems more appropriate.

File: packages/wallet/app/components/shared/animations/BaseAnimation.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    const { animationProgress } = this.state;
+
+    Animated.timing(animationProgress, {
+      duration: 300,
```
Comment (arjunu) [inline packages/wallet/app/components/shared/animations/BaseAnimation.tsx:69]: can you move these to constants?
SHOW_ANIMATION_DURATION
HIDE_ANIMATION_DURATION

File: packages/wallet/app/components/shared/animations/BaseAnimation.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    const renderProp = () => <Text testID="content">Animation content</Text>;
+    const { queryByTestId, rerender } = render(
+      <BaseAnimation isActive={true} render={renderProp} />,
+    );
+
+    rerender(<BaseAnimation isActive={false} render={renderProp} />);
+
+    expect(queryByTestId("content")).toBeDefined();
+
+    timeTravel(200);
```
Comment (arjunu) [inline packages/wallet/app/components/shared/animations/BaseAnimation.spec.tsx:44]: timeTravel(HIDE_ANIMATION_DURATION)

File: packages/wallet/app/modules/signer-ui/reducer.ts (hunk 1 above)
Comment (desfero) [reply packages/wallet/app/modules/signer-ui/reducer.ts:7]: no sure now let's see how the enum evolves when we add support for transaction from the wallet app

File: packages/wallet/storybook/stories.ts (hunk 2 above)
Comment (desfero) [reply packages/wallet/storybook/stories.ts:7]: wow i'm curious why it was not caught by typescript compiler

File: packages/wallet/app/components/shared/animations/BaseAnimation.tsx (hunk 4 above)
Comment (desfero) [reply packages/wallet/app/components/shared/animations/BaseAnimation.tsx:69]: Even better, moved to props so there is control over how long animation should be in progress

File: packages/wallet/app/components/shared/animations/BaseAnimation.spec.tsx (hunk 5 above)
Comment (desfero) [reply packages/wallet/app/components/shared/animations/BaseAnimation.spec.tsx:44]: done

**Pull request merged**